shub_image/deploy.py

```diff
@@ -125,9 +127,12 @@ def _prepare_deploy_params(project, version, image_name, endpoint, apikey,
     if not username:
         params['pull_insecure_registry'] = True
     else:
+        # Missing password leads to auth request to registry authentication
+        # service without 'account' query parameter which breaks login
+        # procedure.
         params['pull_auth_config'] = json.dumps(
             {'username': username,
-             'password': password,
+             'password': password or ' ',
```

Comment:
@vshlapakov so here when user uses private docker hub image and specifies username but does not specify password - it's silently replaced with space? so far I doesn't look like a good idea because it's acceptable only for our internal registry. what do you think about the following - we make password field required if `--username` is set (and vice versa). and for internal registry we add single `--apikey` argument.